### PR TITLE
Multibatch

### PR DESCRIPTION
**Pull request by ElleryBaines**

### CODE REVIEW

File: plaidbench.py [hunk 1]
```diff
@@ -257,30 +260,65 @@ def main():
...
+                              output, x_train, examples, stop_watch)
+
+                # Record stopwatch times
+                execution_duration = stop_watch.elapsed()
+                compile_duration = compile_stop_watch.elapsed()
+
+                network_data['compile_duration'] = compile_duration
+                network_data['execution_duration'] = execution_duration / examples
+                network_data['precision'] = output.precision
+                network_data['example_size'] = examples
```
Comment (earhart) [inline plaidbench.py]: Could it be useful to attach the batch size in here?

File: plaidbench.py [hunk 2]
```diff
@@ -257,30 +260,65 @@ def main():
...
+                # Print statements
+                printf(ex)
+                printf('Set --print-stacktraces to see the entire traceback')
 
-            # Loading the model
-            module, x_train, model = load_model(args.module, x_train)
+                # Record error
+                network_data['exception'] = str(ex)
+                
+                # Set new exist status
```
Comment (earhart) [inline plaidbench.py]: "exit" status

File: plaidbench.py [hunk 3]
```diff
@@ -257,30 +260,65 @@ def main():
...
-            else:
-                # inference run
-                inference(args.module, model, batch_size, compile_stop_watch, 
-                          output, x_train, examples, stop_watch)
+                # stacktrace loop
+                if args.print_stacktraces:
+                    raise NotImplementedError                
 
             # Record stopwatch times
             execution_duration = stop_watch.elapsed()
```
Comment (earhart) [inline plaidbench.py]: I don't think execution_duration is being used here anymore.

File: plaidbench.py [hunk 4]
```diff
@@ -257,30 +260,65 @@ def main():
...
+                    inference(args.module, model, batch_size, compile_stop_watch, 
+                              output, x_train, examples, stop_watch)
+
+                # Record stopwatch times
+                execution_duration = stop_watch.elapsed()
+                compile_duration = compile_stop_watch.elapsed()
+
+                network_data['compile_duration'] = compile_duration
+                network_data['execution_duration'] = execution_duration / examples
+                network_data['precision'] = output.precision
```
Comment (earhart) [inline plaidbench.py]: I think you can omit `precision` and `example_size` -- they're being set later.

File: plaidbench.py [hunk 5]
```diff
@@ -210,6 +210,7 @@ def main():
         parser.add_argument('module', choices=SUPPORTED_NETWORKS)
     args = parser.parse_args()
 
+
```
Comment (brianretford) [inline plaidbench.py]: remove extra blank line

File: plaidbench.py [hunk 6]
```diff
@@ -257,30 +260,65 @@ def main():
         outputs['run_configuration'] = data.copy()
     else:
         networks.append(args.module)
+        batch_list.append(args.batch_size)
 
     for network in networks:
         printf("\nCurrent network being run : " + network)
```
Comment (brianretford) [inline plaidbench.py]: print batch size now also

File: plaidbench.py (hunk 1 above)
Comment (brianretford) [reply plaidbench.py]: Definitely need batch size in the report

File: plaidbench.py (hunk 6 above)
Comment (brianretford) [inline plaidbench.py]: Remove this and combine with something like:

`"Running {0} examples of {1}, batch size {2}".format(args.examples, network, batch)`

**Comment by ElleryBaines** [comment]
re-submitted with corrections from comments

File: plaidbench.py [hunk 7]
```diff
@@ -98,13 +98,13 @@ def train(x_train, y_train, epoch_size, model, batch_size, compile_stop_watch,
     stop_watch.start_outer()
     
     run_initial(batch_size, compile_stop_watch, network, model)
-    model.train_on_batch(x_train[0:batch_size], y_train[0:batch_size])
+    Smodel.train_on_batch(x_train[0:batch_size], y_train[0:batch_size])
```
Comment (brianretford) [inline plaidbench.py]: Smodel is wrong, definitely test completely

File: plaidbench.py [hunk 8]
```diff
@@ -257,79 +260,86 @@ def main():
...
+                if args.print_stacktraces:
+                    raise NotImplementedError                
+
+            # stores network data in dictionary
+            if args.blanket_run:
+                composite_str = network + " : " + str(batch_size)
+                outputs[composite_str] = dict(network_data)
+            # write all data to result.json / report.npy if single run
+            else:
+                network_data['model'] = network
```
Comment (brianretford) [inline plaidbench.py]: move this above where all the other network_data settings are (i.e., set this for blanket too)

**Comment by ElleryBaines** [comment]
ping for review

**Comment by ElleryBaines** [comment]
ping for review

**Comment by ElleryBaines** [comment]
ping for review

**Pull request merged**